### DISCUSSION TITLE
ship distance in metric

### DISCUSSION THREAD

**Discussion by OE1MWW** [jvde-github/AIS-catcher#150] [General]
Hi,
Thanks for that fine software!
I am monitoring in central Europe, AIS-catcher does it flawless.
There is a plugin to convert ships dimensions to feet. Disabled it,
now I see demensions in meter, we are used to metric dimensions.

Unfortunately the distance calculation (I use Lat / Long) is still in 'nmi'
How can this be changed ?

greetings from Austria / Europe

**Comment by OE1MWW** [comment]
my own solution - it works! ;-)

plugin named "distance_in_km.pjs"

// {"author":"OE1MWW","description":"displays unit of distance in km"}

getDistanceVal = c => Number(c * 1.852).toFixed(0);
getDistanceUnit = c => "km";

**Comment by jvde-github** [comment]
Yes, perfect! That is the way to do it.

**Comment by Taxom** [comment]
Since the topic of units of measurement has come up, can I ask here too? How to change the bar scale at the bottom of the maps so that there are kilometres and nautical miles instead of statute ones.
![изображение](https://github.com/jvde-github/AIS-catcher/assets/20313019/725c7b8f-7df3-4977-be2c-028c94a5e21d)

**Reply by jvde-github** [reply]
Looks like in Leaflet you have to write a custom scale class for this.... Will have a look whether there is a simpler solution and then perhaps can be included in a plugin.
